crates/bevy_ecs/src/bundle.rs

```diff
@@ -716,6 +724,37 @@ impl Bundles {
         // SAFETY: index either exists, or was initialized
         unsafe { self.bundle_infos.get_unchecked(id.0) }
     }
+
+    /// Initializes a new dynamic [Bundle]
+    ///
+    /// Dynamic bundles are not cached and each call to
+    /// [`Bundles::init_dynamic_info`] will instantiate a new [`BundleInfo`]
+    /// entry.
+    ///
+    /// The user should cache the returned [`BundleInfo`] as is appropriate for
+    /// their needs.
+    pub fn init_dynamic_info<'a>(
```

Comment:
This method needs a # Panics section in its docs, as this is critical information for using it correctly.

Reply:
Done

Reply:
That's a good point, this isn't in the hot path anywhere so might as well cache this, will fix.